plugins/flatpak/gs-flatpak-transaction.c

```diff
@@ -62,6 +62,52 @@ gs_flatpak_transaction_set_no_deploy (FlatpakTransaction *transaction, gboolean
 	g_object_notify (G_OBJECT (self), "no-deploy");
 }
 
+/* Sets installed app(s) back to installed state. Since libflatpak can return apps as updatable
+ * for re-installing a runtime if its is detected that the runtime was missing at the first place.
+ *
+ * We can determine whether is GsApps is only being updated due to a missing runtime, by checking
```

Comment:
What about the case where an installed app has an update available, and its runtime also has an update available? Are we prematurely setting the app to the installed state in that case?

Reply:
Good to think more on cases like this. 
On top of my head, runtime is missing at the first place in these scenarios, hence having runtime-update seems unlikely.

Reply:
Oh, this function is only called if the operation is an install, not an update, so the situation I asked about isn't relevant. But what about when an app and its runtime are being installed? Wouldn't we prematurely set the app to `AS_APP_STATE_INSTALLED`, since the runtime operation is ordered before the app one?

Reply:
It should not set the app state prematurely for  (app+runtime) install 
a) runtimes are installed before the app, no? 
b) we only set the state iff the app is deployed at the first place. `flatpak_installed_ref_load_metadata` checks if the ref has a deploy file on the installation or not

Reply:
Good point, I forgot about the `flatpak_installed_ref_load_metadata()` call.